OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -137,6 +138,9 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			target += dir * info.BeyondTargetRange.Length / 1024;
 
 			length = Math.Max((target - headPos).Length / speed.Length, 1);
+
+			var rangeModifiers = args.SourceActor.TraitsImplementing<IRangeModifier>().Select(a => a.GetRangeModifier());
+			weaponRange = new WDist(Util.ApplyPercentageModifiers(args.Weapon.Range.Length, rangeModifiers));
```

Comment:
Can't we use the modifiers from the projectile args?
```suggestion
			weaponRange = new WDist(Util.ApplyPercentageModifiers(args.Weapon.Range.Length, args.RangeModifiers));
```

Reply:
Aha, yep much cleaner.